Tests/MessagingPush/MessagingPushTest.swift

```diff
@@ -0,0 +1,156 @@
+@testable import CioTracking
+@testable import CioMessagingPush
+import Foundation
+import SharedTests
+import XCTest
+
+class MessagingPushTest: UnitTest {
+    private var mockCustomerIO: CustomerIO!
+    private var messagingPush: MessagingPush!
+
+    private var identifyRepositoryMock: IdentifyRepositoryMock!
+    
+    private var httpClientMock: HttpClientMock!
+
+    override func setUp() {
+        super.setUp()
+
+        httpClientMock = HttpClientMock()
+        identifyRepositoryMock = IdentifyRepositoryMock()
+
+        mockCustomerIO = CustomerIO(credentialsStore: nil, sdkConfig: SdkConfig(), identifyRepository: identifyRepositoryMock,
+                                keyValueStorage: nil)
+        
+        mockCustomerIO.setCredentials(siteId: String.random, apiKey: String.random, region: Region.EU)
+        
+        messagingPush = MessagingPush(customerIO: mockCustomerIO, httpClient: httpClientMock, keyValueStorage: DITracking.shared.keyValueStorage)
+    }
+
+    // MARK: registerDeviceToken
```

Comment:
I would write a test for the scenario of: http error happens - assert device token not saved by the SDK.

Reply:
test_registerDeviceToken_givenHttpFailure_expectNilDeviceToken